### PR TITLE
feat(extension,common,core): LW-10193 LW-10332 LW-10073 revamp multi-wallet restore and create flows

### PR DESCRIPTION
**Pull request by szymonmaslowski**
# Checklist

- [x] JIRA - [LW-10193](https://input-output.atlassian.net/browse/LW-10193) [LW-10332](https://input-output.atlassian.net/browse/LW-10332) [LW-10073](https://input-output.atlassian.net/browse/LW-10073) [LW-10392](https://input-output.atlassian.net/browse/LW-10392)
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

- Updated multi-wallet create and restore flows to reflect onboarding
  - proper screen order
  - no old redundant screens
  - proper timeline on the left-hand side
  - proper translations
- Added translation for agreement text displayed at the bottom of the onboarding home page
- Added `data-testid` attributes for the recovery phrase length buttons

## Testing

Create and restore flows should reflect those from onboarding. There might be small differences though. Known differences are:
 - in create flow when you're on the 3rd (name and password) step, when you click back you land on the 1st step (mnemonic writedown)

## Screenshots

image
image
image
image

[LW-10193]: https://input-output.atlassian.net/browse/LW-10193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LW-10332]: https://input-output.atlassian.net/browse/LW-10332?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LW-10073]: https://input-output.atlassian.net/browse/LW-10073?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LW-10392]: https://input-output.atlassian.net/browse/LW-10392?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicVerificationStepRevamp.tsx [hunk 1]
```diff
@@ -43,7 +42,15 @@ export const WalletSetupMnemonicVerificationStepRevamp = ({
   const description = (
     <>
       {translations.enterPassphraseLength}
-      <Segmented options={MNEMONIC_LENGTHS} defaultValue={defaultMnemonicLength} onChange={onSetMnemonicLength} />
+      <Segmented
+        options={[
```
Comment (shawnbusuttil) [inline packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicVerificationStepRevamp.tsx:46]: Map? :)

File: apps/browser-extension-wallet/src/lib/translations/en.json [hunk 2]
```diff
@@ -1289,6 +1288,7 @@
   "core.walletSetupMnemonicVerificationStepRevamp.enterPassphrase": "Enter your recovery phrase",
   "core.walletSetupOptionsStep.title": "Let's explore Web3 together",
   "core.walletSetupOptionsStep.subTitle": "Choose an option to get started",
+  "core.walletSetupOptionsStep.agreementText": "By clicking the Create, Connect or Restore button above, you agree with Lace’s <a1>Terms of Service</a1> and <a2>Privacy Policy</a2>",
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/lib/translations/en.json:1291]: Knowledge sharing: where is this markup from?

Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/lib/translations/en.json:1291]: @greatertomi I think this conflicts with your change?

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/RestoreRecoveryPhrase.tsx [hunk 3]
```diff
@@ -1,98 +1,50 @@
...
-      data.mnemonic[i] = '';
-    }
-    data.password = '';
-  }, [data]);
+  }, [mnemonicLength, setMnemonic]);
 
   const walletSetupMnemonicStepTranslations = {
-    writePassphraseTitle: t('core.walletSetupMnemonicStepRevamp.writePassphraseTitle'),
     enterPassphrase: t('core.walletSetupMnemonicStepRevamp.enterPassphrase'),
-    enterPassphraseDescription: t('core.walletSetupMnemonicStepRevamp.enterPassphraseDescription'),
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/RestoreRecoveryPhrase.tsx:49]: We don't need these translations?

**Review by shawnbusuttil — COMMENTED** [review]
Left some initial comments.

File: packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicVerificationStepRevamp.tsx (hunk 1 above)
Comment (szymonmaslowski) [reply packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicVerificationStepRevamp.tsx:46]: Do you mean to generate the array content with an array.map? Good idea :+1:

File: apps/browser-extension-wallet/src/lib/translations/en.json (hunk 2 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/lib/translations/en.json:1291]: It's provided with the `components` property [here](https://github.com/input-output-hk/lace/pull/1100/files#diff-acd80c2370fb4b7099c459b9e82a30ca629265698994e2b3bbc68e887dec46e7R59). Docs: https://react.i18next.com/latest/trans-component#alternative-usage-which-lists-the-components-v11.6.0

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/RestoreRecoveryPhrase.tsx (hunk 3 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/RestoreRecoveryPhrase.tsx:49]: The removed translations are related to the create flow. The `WalletSetupMnemonicVerificationStepRevamp` component does not expect them.

File: apps/browser-extension-wallet/src/lib/translations/en.json (hunk 2 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/lib/translations/en.json:1291]: I've made a comment about that: https://github.com/input-output-hk/lace/pull/1094#discussion_r1579111393

File: packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicVerificationStepRevamp.tsx (hunk 1 above)
Comment (shawnbusuttil) [reply packages/core/src/ui/components/WalletSetupRevamp/WalletSetupMnemonicStepRevamp/WalletSetupMnemonicVerificationStepRevamp.tsx:46]: Yes

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx [hunk 4]
```diff
@@ -25,41 +37,65 @@ export const useCreateWallet = (): State => {
 };
 
 export const CreateWalletProvider = ({ children, providers }: Props): React.ReactElement => {
-  const [state, setState] = useState<Data>({
-    mnemonic: providers.generateMnemonicWords(),
-    name: '',
-    password: ''
+  const history = useHistory();
+  const { clearSecrets, createWallet, createWalletData, setCreateWalletData } = useSoftwareWalletCreation({
+    initialMnemonic: providers.generateMnemonicWords(),
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx:42]: Why 'SoftwareWallet'? 🤔

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx [hunk 5]
```diff
@@ -1,18 +1,30 @@
...
+import { Providers } from './types';
 
 interface Props {
   children: React.ReactNode;
   providers: Providers;
 }
 
+type OnNameAndPasswordChange = (state: { name: string; password: string }) => void;
+
+enum Step {
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx:16]:
```suggestion
enum WalletSetupStep {
```

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/context.tsx [hunk 6]
```diff
@@ -25,43 +40,76 @@ export const useRestoreWallet = (): State => {
...
+  const analytics = useAnalyticsContext();
+  const [step, setStep] = useState<Step>(Step.RecoveryPhrase);
+  const {
+    clearSecrets,
+    createWallet: createSoftwareWallet,
+    createWalletData,
+    setCreateWalletData
+  } = useSoftwareWalletCreation({
+    initialMnemonic: [],
+    postHogActionWalletAdded: PostHogAction.MultiWalletRestoreAdded
```
Comment (shawnbusuttil) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/context.tsx:53]: Do we have to pass posthog params like this? Can we not call the event as a side effect the same way we called `analytics` after we take an action?

**Review by shawnbusuttil — DISMISSED** [review]
Left some more comments.

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx (hunk 4 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx:42]: We have one flow for the Hardware Wallet and two flows for the Software Wallet, right? An alternative might be `useInMemoryWalletCreation`. Do you have a suggestion?

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/context.tsx (hunk 6 above)
Comment (szymonmaslowski) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/context.tsx:53]: So basically this comes down to whether to reduce the duplication of the following code which is the same for the two flows:
```
    const result = await walletManager.createWallet(createWalletData);
    void analytics.sendEventToPostHog(postHogActionWalletAdded, {
      // eslint-disable-next-line camelcase
      $set: { wallet_accounts_quantity: await getWalletAccountsQtyString(walletManager.walletRepository) }
    });
    void analytics.sendMergeEvent(result.source.account.extendedAccountPublicKey);
```

IMO it was worth so I exposed it as a `createWallet` function. I don't really have a strong opinion here, so If you think this code should be repeated, I am happy to change it.

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx (hunk 4 above)
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/create-wallet/context.tsx:42]: Hot Wallet?

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/context.tsx (hunk 6 above)
Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/context.tsx:53]: I think the duplication here is minimal. Not too keen on passing concrete analytics types into functions.

File: apps/browser-extension-wallet/src/lib/translations/en.json (hunk 2 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/lib/translations/en.json:1311]: I removed this change from my PR already.

File: apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/Setup.tsx [hunk 7]
```diff
@@ -22,22 +21,33 @@ export const Setup = (): JSX.Element => {
...
+      } else {
+        throw error;
+      }
+    }
+
+    await next();
+  };
+
   return (
-    <WalletSetupNamePasswordStep
```
Comment (greatertomi) [inline apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/Setup.tsx:31]: Should this component and all the other unused ones like `MnemonicWordsWritedown`  be deleted in this PR or on a subsequent one?

Comment (shawnbusuttil) [reply apps/browser-extension-wallet/src/views/browser-view/features/multi-wallet/restore-wallet/steps/Setup.tsx:31]: The component you mentioned is still being used in settings.

Any other cleaning up potentially in: https://input-output.atlassian.net/browse/LW-10194

**Review by shawnbusuttil — APPROVED** [review]
Good job. 🚀

**Review by greatertomi — APPROVED** [review]

**Pull request merged**